Source/WTF/wtf/NativePromise.h

```diff
@@ -368,6 +368,16 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
 
 private:
     friend class Producer;
+    class Storage;
+
+    template<typename SettleValueType>
+    inline void settleImpl(SettleValueType&& result, Locker<Lock>& lock)
+    {
+        assertIsHeld(m_lock);
+        m_result.optionalResult() = std::forward<SettleValueType>(result);
```

Comment:
Suggestion:
`std::optional::operator=` is quite "wide", is that it can handle lots of cases, like copy-assignment (from the same type of optional); and depending on whether the optional is already engaged it may do either an initialization or an assignment; etc.
But I believe this code here should only be used once, to set a value in an empty optional, right?
So I would suggest using `emplace`, which just focuses on initializing the contained type with the given argument.

Reply:
You could move the isNothing() ASSERT here, or make m_result have an initialise method that would ASSERT it is not already initialised.